### ISSUE TITLE
Error Details with AWS S3

### ISSUE THREAD

**Issue by rahulguptajss** [dataversioncontrol/dvc#616]
Hi,

For any AWS S3 related issue, I get generic error message for all type of issues. Can we make it more explanatory? I mean if it is authentication/missing permissions etc.

Failed to push data to the cloud: Data sync error: Storage path ml-accelerator is not setup correctly

**Comment by efiop** [comment]
Thanks for the feedback! I agree, the error msgs for our clouds are not very informative. I'm currently reworking our configs(including cloud configs) and this issue should be resolved in an upcoming release.

**Comment by efiop** [comment]
Btw, looks like I get where you got this error from. I can confirm that I encounter it too. It is a regression since ~0.9.0. Looking into solving it.

**Comment by mbatchkarov** [comment]
I agree that the messages could be more useful, but it was fairly straightforward to debug the issue.  In case this helps anyone, here is what I did:

First, open `.dvc/config` and check all settings:

 - Is the cloud provider set (in my case that's AWS)
 - Does the credentials file exist?
 - Does your S3 bucket exist?
 - Do you have the right credentials in the aws config file? Can you use `awscli` to list the bucket, e.g. `aws s3 ls my-dev-bucket`?
 - Have you set the correct S3 region? For some reason `dvc`/`boto` ignores the setting from `~/.aws/config`, so you have to explicitly add the to the dvc conf file


Here is my configuration file for reference:

```
[Core]
Cloud = AWS

 [AWS]
CredentialPath = ~/.aws/credentials
Profile = dev
Region = eu-central-1
StoragePath = my-dev-bucket/tutorial
```

**Comment by efiop** [comment]
@mbatchkarov Thank you for the detailed walk-through!

> For some reason dvc/boto ignores the AWS_REGION setting from ~/.aws/config

I'll take a look into it. Created https://github.com/dataversioncontrol/dvc/issues/636

**Comment by efiop** [comment]
We've migrated to boto3 on current master, thus allowing us to only specify the desired profile for  the credentials(boto3 will use default profile if it is not specified in `profile` option in our config) and nothing more, meaning that if you have `aws s3` tool working, `dvc` should work too. Looks like this is the best behavior that we can achieve.

Also, with new `dvc remote` concept all you need to do to add s3 cloud is:
```
dvc remote add my-s3-cloud s3://bucket/dir
dvc config core.remote my-s3-cloud #this so that you dvc will now what remote to use by default
```

These changes will be released in 0.9.6. Closing this for now.
